### PR TITLE
Adding API revision feature migration steps

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Issues

https://github.com/wso2/product-apim/issues/9406

### CODE REVIEW

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/dao/APIMgtDAO.java [hunk 1]
```diff
@@ -552,4 +555,33 @@ public void updateUUID(List<APIInfoDTO> apiInfoDTOS) throws APIMigrationExceptio
...
+    /**
+     * Get all APIs
+     *
+     * @return List of APIInfoDTO objects
+     */
+    public static List<APIInfoDTO> getAPIs() throws APIMigrationException {
+        List<APIInfoDTO> allAPIs = new ArrayList<>();
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            connection.setAutoCommit(false);
+            try (PreparedStatement preparedStatement = connection.prepareStatement(GET_ALL_APIS)){
```
Comment (wasuradananjith) [inline migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/dao/APIMgtDAO.java:568]:
```suggestion
            try (PreparedStatement preparedStatement = connection.prepareStatement(GET_ALL_APIS)) {
```

Comment (wasuradananjith) [reply migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/dao/APIMgtDAO.java:568]: Shall we add a space shown above? Also, there are several places like this. Shall we fix those as well?

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java [hunk 2]
```diff
@@ -105,4 +132,114 @@ public void spMigration() throws APIMigrationException {
...
+                    String apiPath = GovernanceUtils.getArtifactPath(registry, apiInfoDTO.getUuid());
+                    int prependIndex = apiPath.lastIndexOf("/api");
+                    String artifactPath = apiPath.substring(0, prependIndex);
+                    String artifactOldPathIcon = APIConstants.API_IMAGE_LOCATION + RegistryConstants.PATH_SEPARATOR
+                            + apiInfoDTO.getApiProvider() + RegistryConstants.PATH_SEPARATOR + apiInfoDTO.getApiName()
+                            + RegistryConstants.PATH_SEPARATOR + apiInfoDTO.getApiVersion()
+                            + RegistryConstants.PATH_SEPARATOR + APIConstants.API_ICON_IMAGE;
+                    if (registry.resourceExists(artifactOldPathIcon)) {
+                        Resource resource = registry.get(artifactOldPathIcon);
+                        String thumbPath = artifactPath + RegistryConstants.PATH_SEPARATOR + APIConstants.API_ICON_IMAGE;
```
Comment (wasuradananjith) [inline migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java:153]:
```suggestion
                        String thumbPath =
                                artifactPath + RegistryConstants.PATH_SEPARATOR + APIConstants.API_ICON_IMAGE;
```

Comment (wasuradananjith) [reply migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java:153]: Shall we split this to two lines? Character length has exceeded

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java [hunk 3]
```diff
@@ -105,4 +132,114 @@ public void spMigration() throws APIMigrationException {
...
+                        } else {
+                            wsdlResourcePath = wsdlResourcePathFile;
+                        }
+                        registry.copy(wsdlResourcePathOld, wsdlResourcePath);
+                        GenericArtifactManager artifactManager = RegistryPersistenceUtil.getArtifactManager(registry,
+                                APIConstants.API_KEY);
+                        GenericArtifact apiArtifact = artifactManager.getGenericArtifact(apiInfoDTO.getUuid());
+                        String absoluteWSDLResourcePath = RegistryUtils
+                                .getAbsolutePath(RegistryContext.getBaseInstance(), RegistryConstants.GOVERNANCE_REGISTRY_BASE_PATH)
+                                + wsdlResourcePath;
```
Comment (wasuradananjith) [inline migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java:185]:
```suggestion
                        String absoluteWSDLResourcePath = RegistryUtils
                                .getAbsolutePath(RegistryContext.getBaseInstance(),
                                        RegistryConstants.GOVERNANCE_REGISTRY_BASE_PATH) + wsdlResourcePath;
```

Comment (wasuradananjith) [reply migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java:185]: Shall we split this to two lines? Maximum character length has exceeded

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/dao/APIMgtDAO.java (hunk 1 above)
Comment (CrowleyRajapakse) [reply migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/dao/APIMgtDAO.java]: resolved

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java (hunk 2 above)
Comment (CrowleyRajapakse) [reply migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java]: resolved

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java (hunk 3 above)
Comment (CrowleyRajapakse) [reply migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java]: resolved

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java [hunk 4]
```diff
@@ -105,4 +132,118 @@ public void spMigration() throws APIMigrationException {
...
             throws APIMigrationException {
     }
+
+    public void updateRegistryPathsOfIconAndWSDL() throws APIMigrationException {
+        List<APIInfoDTO> apiInfoDTOList;
+        try {
+            apiInfoDTOList = apiMgtDAO.getAPIs();
+            if (!apiInfoDTOList.isEmpty()) {
+                this.registry = ServiceReferenceHolder.getInstance().getRegistryService().
+                        getGovernanceUserRegistry("admin", MultitenantConstants.SUPER_TENANT_ID);
```
Comment (tharindu1st) [inline migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java:142]: use tenant admin instad of hardcoded "admin"

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java (hunk 4 above)
Comment (CrowleyRajapakse) [reply migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/client/MigrateFrom320.java]: resolved

**Review by chamilaadhi — APPROVED** [review]

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**